types.go

```diff
@@ -14,6 +14,58 @@ const (
 	nullBody                 = "null"
 )
 
+// Network represents the Meilisearch network configuration.
+// Each field is wrapped in an Opt so it can be explicitly included,
+// set to JSON null, or omitted entirely.
+type Network struct {
+	Self    Opt[string]                 `json:"self,omitempty"`
+	Remotes Opt[map[string]Opt[Remote]] `json:"remotes,omitempty"`
+}
+
+func (n Network) MarshalJSON() ([]byte, error) {
```

Comment:
Optional does not require writing a custom marshal or unmarshal, since it already provides specific implementations for both.

https://github.com/meilisearch/meilisearch-go/blob/21c87562c26c8f04a5f69d4d4d3bbc7db0769372/field.go#L66-L82